sig-network/link-state-proposal.md

```diff
@@ -0,0 +1,197 @@
+# VEP #2: Adding Link State Management for vNICs
+
+## Release Signoff Checklist
+Items marked with (R) are required *prior to targeting to a milestone / release*.
+
+- [ ] (R) Enhancement issue created, which links to VEP dir in [kubevirt/enhancements] (not the initial VEP PR)
+
+## Overview
+
+The purpose of this enhancement is to introduce support for link state management (up/down) in KubeVirt. This feature
+will enable VM owners to dynamically control the link state. It aligns KubeVirt with traditional virtualization platforms
+and enhances its utility in environments requiring precise network state control.
+
+By introducing link state management, KubeVirt will:
+- Provide VM-level controls to configure the link state of individual interfaces.
+- Report link state info via the VirtualMachineInterface status.
+
+This proposal outlines the necessary changes to KubeVirt, including API changes.
+
+## Motivation
+
+Currently, KubeVirt does not support modifying or reporting the link state of virtual network interfaces.
+This limitation can lead to challenges in implementing advanced networking scenarios, such as:
+- Simulating link failures for testing and debugging.
+- Managing multi-network setups, where some interfaces need to be intentionally disabled.
+
+## Goals
+
+1. Allow users to configure the link state (up/down) of individual interfaces through the KubeVirt API.
+2. Enable link state configuration at both VM creation and runtime.
+3. Report the current link state of each VM interface.
+
+## Non Goals
+
+1. Affect network plumbing on the host - the setup process will remain unchanged.
+
+## User Stories
+
+As a VM owner I would like to be able to:
+
+- Start the VM with one or more interfaces with their link set to `down`.
+- Toggle the link state of one or more interfaces of a running VM.
+- Hot plug an interface with a link state set to `down`.
+
+- I want my interfaces link states to persist following a migration.
+
+## Repos
+
+kubevirt/kubevirt.
+
+## Design
+
+### API Addition
+Currently, the `Interface` struct has the `State` field (currently used for hot-unplug):
+```go
+type Interface struct {
+    ...
+    State InterfaceState `json:"state,omitempty"`
+    ...
+}
+```
+
+Type `InterfaceState` currently has a single allowed value:  
+```go
+type InterfaceState string
+
+const (
+	InterfaceStateAbsent InterfaceState = "absent"
+)
+```
+
+The following values will be added to specify the required link state:
+```go
+const (
+    InterfaceStateLinkUp   InterfaceState = "up"
+    InterfaceStateLinkDown InterfaceState = "down"
+)
+```
+
+An empty value will be considered as `up`.
+
+The network validator will be adjusted to allow the two new values for all interfaces regardless of whether they are 
+connected to primary or secondary networks, or whether they use a core binding or a binding plugin.
```

Comment:
1. Setting A VM's masquerade (pod) interface to `down` - is it even legit? How will it actually affect the guest-to-pod connectivity?
2. In UDN VM - will setting the primary interface to `down` only affect the node-facing part of the interface, and keep the pod networking fully functioning?

Reply:
Which according to [1]:
> If 'down' is specified as the value, the interface behaves as if it had the network cable disconnected.

[1] https://libvirt.org/formatdomain.html#modifying-virtual-link-state